kolena/_experimental/object_detection/utils.py

```diff
@@ -267,3 +267,40 @@ def compute_optimal_f1_threshold_multiclass(
     for label, (y_true, y_score) in y_true_score_by_label.items():
         optimal_thresholds[label] = _compute_optimal_f1_with_arrays(y_true, y_score)
     return optimal_thresholds
+
+
+def compute_average_precision(precisions: List[float], recalls: List[float]) -> float:
+    """
+    Computes the average precision given a PR curve with the metrics methodology of PASCAL VOC.
+    Based on the [PASCAL VOC code in Python](https://github.com/Cartucho/mAP).
+
+    :param precisions: A list precision values from a PR curve.
+    :param recalls: A list recall values from a PR curve.
+    :return: The value of the average precision.
+    """
+
+    if len(precisions) != len(recalls):
+        raise ValueError("precisions and recalls differ in length")
+
+    pairs = sorted(zip(recalls, precisions), key=lambda x: x[0])
+    recalls = [x[0] for x in pairs]
+    precisions = [x[1] for x in pairs]
```

Comment:
This isn't important, but for readability it's a good practice to use destructuring where possible. Here that would mean:

```python
recalls = [recall for recall, _ in pairs]
precision = [precision for _, precision in pairs]
```